Classes/Service/OgRendererService.php

```diff
@@ -140,7 +140,18 @@ public function main($content, $conf)
             $og['image'] = $fileObjects;
 
         // Get url
-        $og['url'] = htmlentities(GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'));
+        $og['url'] = GeneralUtility::getIndpEnv('TYPO3_SITE_URL');
+        $cObject = GeneralUtility::makeInstance('tslib_cObj');
+        $configurations['additionalParams'] = "&L=".(int)GeneralUtility::_GP('L');
+        $configurations['returnLast'] = 'url'; // get it as URL
+        $configurations['parameter'] =  $GLOBALS['TSFE']->id;
```

Comment:
That's right.
I'm going to enhance and test your proposed solution and give some feedback. I'll try to get some time for this this weekend, otherwise on monday.